scalatest/src/main/scala/org/scalatest/Entry.scala

```diff
@@ -48,14 +48,14 @@ case class Entry[K, V](key: K, value: V) extends java.util.Map.Entry[K, V] {
    *
    * @return the key corresponding to this entry
    */
-  def getKey: K = key
+  def getKey: K = key.asInstanceOf[K]
```

Comment:
**Overriden java method return type cannot be null**
The nullness in the return type of all java methods are cast away by the `-Yjava-interop-dont-nullify-outermost` compiler flag. `asInstanceOf[A]` is used for return values of type `A | Null`.